### PR TITLE
Visual Studio 2015 Support

### PR DESCRIPTION
**Pull request by fuzun**
- Did not remove msvc6 project files or old batch scripts.
- Only made required changes for vs2015 and for some high level pvs studio warnings.
- PVS studio spotted some functions that can lead overflow. I will look for these at another branch, as this is only for VS2015.
- Had to change HSPRITE. VS2015 did not even like new define macros added recently.

### CODE REVIEW

File: cl_dll/view.cpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ extern cvar_t	*cl_forwardspeed;
 extern cvar_t	*chase_active;
 extern cvar_t	*scr_ofsx, *scr_ofsy, *scr_ofsz;
 extern cvar_t	*cl_vsmoothing;
-extern Vector   dead_viewangles;
+/*extern */ Vector   dead_viewangles;
```
Comment (a1batross) [inline cl_dll/view.cpp]: Are you sure? Then it's definition should be replaced by declaration in ```input_xash3d.cpp```(e.g. add extern key).

I think same fix(as for input_xash3d.cpp) must be applied for inputw32.cpp.

File: cl_dll/view.cpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ extern cvar_t	*cl_forwardspeed;
 extern cvar_t	*chase_active;
 extern cvar_t	*scr_ofsx, *scr_ofsy, *scr_ofsz;
 extern cvar_t	*cl_vsmoothing;
-/*extern */ Vector   dead_viewangles;
+extern  Vector   dead_viewangles;
```
Comment (a1batross) [inline cl_dll/view.cpp]: Ah, you returned it back. :)

Comment (fuzun) [inline cl_dll/view.cpp]: I think we need a proper base where we can see changes. I realized that file very lately. Seeking through all of those non descripted commits isn't a good thing to do :)

File: common/mathlib.h [hunk 3]
```diff
@@ -27,7 +27,7 @@ typedef vec_t vec4_t[4];	// x,y,z,w
 
 struct mplane_s;
 
-extern vec3_t vec3_origin;
+extern vec3_t vec3_origin_c;
```
Comment (fuzun) [inline common/mathlib.h]: Alfred sdk uses some definitions as a workaround I think. But this is really messed up and I did not get the compiler happy without changing it to this.

File: dlls/explode.cpp [hunk 4]
```diff
@@ -164,11 +164,11 @@ void CEnvExplosion::Use( CBaseEntity *pActivator, CBaseEntity *pCaller, USE_TYPE
 	{
 		pev->origin = tr.vecEndPos + ( tr.vecPlaneNormal * ( m_iMagnitude - 24 ) * 0.6 );
 	}
-	else
+	/*else
 	{
-		pev->origin = pev->origin;
-	}
-
+		pev->origin = pev->origin; // ?
```
Comment (fuzun) [inline dlls/explode.cpp]: What is this?

File: cl_dll/in_defs.h [hunk 5]
```diff
@@ -17,9 +17,7 @@
 #define	ROLL	2 
 
 #ifdef _WIN32
-#define HSPRITE HSPRITE_win32
```
Comment (fuzun) [inline cl_dll/in_defs.h]: Does not work for VS2015.

Comment (mittorn) [inline cl_dll/in_defs.h]: It does not build with mingw without it

Comment (fuzun) [inline cl_dll/in_defs.h]: I have noticed it while checking latest commits. Since HSPRITE changed to SpriteHandle_t, it is now ok to add that definition.

**Comment by mittorn** [comment]
i'll check this few days later

**Comment by fuzun** [comment]
CMake lists added so there is no need for this.

**Pull request closed without merging**